cmd/flow-dps-indexer/main.go

```diff
@@ -74,8 +74,9 @@ func run() int {
 	}
 	log = log.Level(level)
 
-	// Open index database.
-	db, err := badger.Open(dps.DefaultOptions(flagIndex))
+	// Open index database and bypass lock guard since the indexer is alone writing on the database and this allows
+	// the server to simultaneously read from the database as it is being indexed.
+	db, err := badger.Open(dps.DefaultOptions(flagIndex).WithBypassLockGuard(true))
```

Comment:
This should not use the `WithBypassLockGuard`; every writer should open the database normally, so that they claim the lock and block any other writers. This would actually allow multiple writers to open the database concurrently.

Reply:
Oh my bad, I thought this was needed in order to allow the others to open the DB in read-only. Makes sense though, will fix, thanks :)